### PR TITLE
Various additions to compile NSPR and NSS

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds some defines and structs to please `NSPR` and `NSS`.

### CODE REVIEW

File: options/bsd/include/sys/queue.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+#define TAILQ_ENTRY(type)	_TAILQ_ENTRY(struct type,)
+
+/*
+ * Tail queue functions.
+ */
+#define	TAILQ_INIT(head) do {						\
+	(head)->tqh_first = NULL;					\
+	(head)->tqh_last = &(head)->tqh_first;				\
+} while (/*CONSTCOND*/0)
```
Comment (Geertiebear) [inline options/bsd/include/sys/queue.h:30]: Remove these ``CONSTCOND`` comments.

**Review by Geertiebear — CHANGES_REQUESTED** [review]

**Comment by avdgrinten** [comment]
The queue implementation seems to be taken from somewhere else. Remember to add licenses!

File: options/bsd/include/sys/queue.h (hunk 1 above)
Comment (Dennisbonke) [reply options/bsd/include/sys/queue.h]: Done as requested.

**Comment by Dennisbonke** [comment]
> The queue implementation seems to be taken from somewhere else. Remember to add licenses!

The header was taken from glibc, but the copyright header in it seems to be a BSD-3-Clause style header, so I think we're good, but I could be wrong.

**Review by Geertiebear — APPROVED** [review]
LGTM now.

**Pull request merged**